pkg/query-service/model/logs.go

```diff
@@ -62,3 +64,38 @@ func GetLogFieldsV3(ctx context.Context, queryRangeParams *v3.QueryRangeParamsV3
 	}
 	return data
 }
+
+type PromotePathsRequest struct {
+	Paths []PromotePathItem `json:"paths"`
+}
+
+type PromotePathItem struct {
+	Path    string `json:"path"`
+	Promote bool   `json:"promote"`
+	Indexed bool   `json:"indexed"`
```

Comment:
set default index if not sent from the user, else let user configure what Index to be sent and how many of them.